Add Knowledge base / staking (#22)
galacticcouncil/HydraDX-docs: docs/staking.md

This section provides a brief introduction into how staking works in the HydraDX network. If you are not experienced in staking in a Substrate-based network, we recommend that you read this before deciding to participate.

The consensus mechanism used by HydraDX is called Nominated Proof-of-Stake (NPoS). NPoS is a variation of Proof-of-Stake and is used in Substrate-based blockchains such as Polkadot and Kusama. The two central actors in an NPoS environment are called [**validators**](#validators) and [**nominators**](#nominators). 

### Validators

Validators participate in the network by running validator nodes which provide the infrastructure allowing a Substrate-based network to operate securely. Validator nodes fulfill three functions which are of paramount importance to the consensus mechanism. In the first place, they validate the information contained in blocks, such as the identity of the parties and the subject of the contract. In the second place, validators participate in the production of new blocks based on the validity statements of other validators. In the third place, they guarantee the finality of blockchain transactions.

> An important characteristic of NPoS is that not all validators participate in the validating process at the same time. Only validators in the *active validator set* perform the above-mentioned operations and earn rewards for doing so. Generally speaking, the set of active validators is limited to a fixed amount of nodes, however in [HydraDX Snakenet](/snakenet) we will pursue a larger set in order to give more people the possibility to participate.

Reviewer: did you clarify this?
my last information was targeting 300first, 900after. 
your text reads like we will have many slots now and a straight cut transitioning to mainnet. this will scare people off

:::note

In a Substrate-based network, time is divided in units called **eras**. In [HydraDX Snakenet](/snakenet), *1 Era = 24 hours*.

:::

Participating as a validator requires a certain level of technical knowledge needed for safely setting up and maintaining a validator node. Misbehavior of a validator node can be punished by slashing, leading to an involuntary loss of funds for you and your nominators. If you think you have the needed experience for running a validator node, you can refer to our [validator guide](/node_setup). Otherwise, we highly recommend that you consider becoming a nominator.

### Nominators

> Nominators help secure the network by nominating validators to be elected in the active validator set. They do so by staking their HDX tokens with validators of their choice. The nomination process does not require running and maintaining nodes which makes this form of staking more accessible to people lacking a strong technical background. Tokens which are used to nominate validators are *bonded*, meaning that they are frozen and cannot be used for other purposes. Nominators can always withdraw their nomination which will be reflected at the end of the current era. Nominators can also free up their tokens, however this will become only effective after a waiting period of *28 days* following the unbonding transaction.

Reviewer: IMHO "lacking" seem a little offending here? maybe it's just me

> Nominators help secure the network by nominating validators to be elected in the active validator set. They do so by staking their HDX tokens with validators of their choice. The nomination process does not require running and maintaining nodes which makes this form of staking more accessible to people lacking a strong technical background. Tokens which are used to nominate validators are *bonded*, meaning that they are frozen and cannot be used for other purposes. Nominators can always withdraw their nomination which will be reflected at the end of the current era. Nominators can also free up their tokens, however this will become only effective after a waiting period of *28 days* following the unbonding transaction.

Reviewer: "Nominators can always withdraw their nomination which will be reflected at the end of the current era"

do you mean withdraw their rewards?
or did you mean unbonding
because process is like 
stop nominating
unbond
(wait unbonding period)
withdraw unbonded

> The choice of validators that you nominate is important for the amount of rewards you are about to receive. You should always do your due diligence before nominating a validator. If your chosen validator fails to live up to the minimum expectations and misbehaves in the network (e.g. too much downtime), it can be punished by slashing which will result in a partial loss of the funds that you have staked with this validator.

Reviewer: for me this reads like you get slashed when the validator doesn't reach a minimum total stake or sth like that
would rephrase that "minimum expectations"

> The choice of validators that you nominate is important for the amount of rewards you are about to receive. You should always do your due diligence before nominating a validator. If your chosen validator fails to live up to the minimum expectations and misbehaves in the network (e.g. too much downtime), it can be punished by slashing which will result in a partial loss of the funds that you have staked with this validator.

Reviewer: "(e.g. too much downtime)"
IMHO downtime is no slashable if it's just this validator. i've seen downtimes slashed only when multiple nodes downed at once
(e.g. do harm or attack the network, security vulnerabilities, multiple validators going offline the same time)

> The first thing you need to consider when choosing a validator is the *reward commission percentage*. This represents the proportion of the rewards which will be paid out to the validator for providing its services to nominators. Furthermore, you should conduct some investigation into the credibility of validators including checking their identity, as well as historical information such as era points, elected stake, rewards and slashes. At the beginning of Snakenet it might be difficult to find all this information - if you are in doubt about the choice of validators, approach as in Discord and we will share with you our community list of trusted validators.

Reviewer 2: I would switch due dilligence as first consideration as it's risky to randomly select validators without it. The commission might be bigger because it costs more for validators with beefier setups, so that is not the best indication how you should select one.

:::note

We strongly recommend that you spread your stake by **nominating several validators**. This enhances the decentralization of the network and increases your chances of electing a trustworthy node in the set of active validators.

> Also, keep in mind that there is a current **maximum of 64 nominations** for a single validator after which it becomes oversubscribed. If your nomination is in excess of this limit, it will be disregarded and you will not earn any rewards for it.

Reviewer: if somebody joins an oversubscribed validator having more tokens than the lowest-stake  active nominator he will be in the game. the nominators with the lowest stake will drop off

---
id: staking
title: Staking
---

import useBaseUrl from '@docusaurus/useBaseUrl';

This section provides a brief introduction into how staking works in the HydraDX network. If you are not familiar with staking in a Substrate-based network, we recommend that you read this before deciding to participate.

The consensus mechanism used by HydraDX is called Nominated Proof-of-Stake (NPoS). NPoS is a variation of Proof-of-Stake and is used in Substrate-based blockchains such as Polkadot and Kusama. The two central actors in an NPoS environment are called [**validators**](#validators) and [**nominators**](#nominators). 

### Validators

> Validators participate in the network by running validator nodes which provide the infrastructure allowing a Substrate-based network to operate securely. Validator nodes fulfill three functions which are of paramount importance to the consensus mechanism. In the first place, they validate the information contained in blocks, such as the identity of the parties and the subject of the contract. In the second place, validators participate in the production of new blocks based on the validity statements of other validators. In the third place, they guarantee the finality of blockchain transactions.

Reviewer 2: ```suggestion
Validators participate in the network by running validator nodes, which provide the infrastructure that allows the HydraDX network to operate securely. Validator nodes fulfill three functions which are of paramount importance to the consensus mechanism. In the first place, they validate the information contained in blocks, such as the identity of the parties and the subject of the contract. In the second place, validators participate in the production of new blocks based on the validity statements of other validators. In the third place, they guarantee the finality of blockchain transactions.
```

An important characteristic of NPoS is that not all validators participate in the validating process at the same time. Only validators in the *active validator set* perform the above-mentioned operations and earn rewards for doing so. The set of active validators is limited to a fixed number of nodes. In [HydraDX Snakenet](/snakenet) we expect this number to be around 300, scaling it up as we progress towards mainnet.

> The decision which validators to elect in the set of active validators is made by following the principle of *proportional justified representation*. This principle aims to safeguard decentralization and fair representation by assigning the available slots to validators in proportion to their nominated stake. The higher the amount of staked tokens with a given validator, the higher the chance that the node will be elected in the active set. Validators which are not included in the active set are placed in a waiting list. The set of active validators is updated at the beginning of every era which provides a possible entry window for new validators.

Reviewer 2: ```suggestion
The decision to select validators in the set of active validators is made by following the principle of *proportional justified representation*. This principle aims to safeguard decentralization and fair representation by assigning the available slots to validators in proportion to their nominated stake. The higher the amount of staked tokens with a given validator, the higher the chance that the node will be elected in the active set. Validators which are not included in the active set are placed on a waiting list. The set of active validators is updated at the beginning of every era, providing a possible entry window for new validators.
```

Reviewer 2: Maybe even rephrase little more first sentence as it's little bit confusing / long.

### Nominators

Nominators help secure the network by nominating validators to be elected in the active validator set. They do so by staking their HDX tokens with the validators of their choice. The nomination process does not require running and maintaining nodes, making this form of staking more accessible to everybody. Tokens which are used to nominate validators are *bonded*, meaning that they are frozen and cannot be used for other purposes. It is at any time possible to change or stop your nominations which will be reflected at the end of the current era. Nominators can also free up their tokens, however this will only become effective after a waiting period of *28 days* following the unbonding transaction.

> Before nominating, you should always do your due diligence and research the credibility of the chosen validators. You can do so by checking their identity as well as historical information like era points, elected stake, rewards and slashes. At the beginning of Snakenet it might be difficult to find all this information - if you are in doubt about the choice of validators, approach us in Discord and we will share with you our community list of trusted validators.

Reviewer 2: ```suggestion
Before nominating, you should always do your due diligence and research the credibility of the chosen validators. You can do so by checking their identity as well as historical information like era points, elected stake, rewards and slashes. At the beginning of Snakenet, it might be difficult to find all this information - if you are in doubt about the choice of validators, approach us in the Discord, and we will share our community curated list of trusted validators with you.
```

Another point to consider when choosing a validator is the *reward commission percentage*. This represents the proportion of the rewards which will be paid out to the validator for providing its services to nominators. The lowest commission percentage is not always the best - running a performant and available node has high operational costs which can only be covered sustainably by demanding a realistic reward commission.

:::note

> We strongly recommend that you spread your stake by **nominating several validators**. This enhances the decentralization of the network and increases your chances of electing a trustworthy node in the set of active validators.

Reviewer: @jak-pan is the stake really spread across several validators? from my polkadot-experience it seemed like you just get elected by one validator.

Reviewer: ```suggestion
We strongly recommend to **nominate several validators**. This enhances the decentralization of the network and increases your chances of being elected by an validator from the active validator set.
```

Reviewer 2: ```suggestion
We strongly recommend to **nominate several validators**. This enhances the decentralization of the network and increases your chances of your stake being included in the elected validator set.
```

> Make sure that you do not nominate validators which are oversubscribed. Currently, there is a **limit of 64 nominations** for a single validator, after which it becomes oversubscribed. When the following era begins, an oversubscribed validator will only be elected using the maximum allowed number of nominations. If this occurs, the nominations with the highest stake take precedence, while the nominations with the lowest stake will be disregarded and will not earn any awards during the era.

Reviewer: ```suggestion
Make sure that you do not nominate validators that are oversubscribed. Currently, there is a **limit of 64 nominations** for a single validator, after which it becomes oversubscribed. When the following era begins, an oversubscribed validator will only be elected using the maximum allowed number of nominators. If this occurs, the nominators with the highest stake take precedence, while the nominators with the lowest stake will be disregarded and will not earn any awards during that era.
```

> Nominating is a more accessible form of staking however it also bears risks. Validators which breach the rules of the network may be punished by slashing, resulting in a partial loss of funds for both the validator and its nominators. This is why it is important to only nominate reputable validator nodes.

Reviewer: ```suggestion
Nominating is a more accessible form of staking however it also bears risks. Validators which breach the rules of the network may be punished by slashing, resulting in a partial to complete loss of funds for both the validator and its nominators. This is why it is important to only nominate reputable validator nodes.
```

:::

> After you have nominated one or more validators, you will not start earning rewards immediately. Before this can happen, the validators in question must receive sufficient nominations and staked tokens to be elected in the set of active validators. Until then, your nomination will remain inactive (for more details see previous section).

Reviewer: ```suggestion
After you have nominated one or more validators, you will not start earning rewards immediately. Before this can happen, the validators in question must receive sufficient staked tokens to be elected in the set of active validators. Until then, your nomination will remain inactive (for more details see previous section).
```

Reviewer 2: sufficient backing?
